content/en/blog/_posts/2021-03-31-kube-state-metrics-goes-v-2-0.md

```diff
@@ -0,0 +1,168 @@
+date: 2021-04-06
+slug: kube-state-metrics-v2-0
+---
+
+**Authors:** Lili Cosic (Red Hat), Frederic Branczyk (Polar Signals), Manuel Rüger
+
+## What?
+
+[Kube-state-metrics](https://github.com/kubernetes/kube-state-metrics), a project under the Kubernetes organization, that generates Prometheus format metrics based on the current state of the Kubernetes native resources. It does this by listening to the Kubernetes API server and gathering information about resources and objects, e.g. Deployments, Pods, Services, and StatefulSets. A full list is available in the [documentation]( https://github.com/kubernetes/kube-state-metrics/tree/master/docs) of kube-state-metrics.
+
+## Why?
+
+After deploying kube-state-metrics it provides you with an array of useful metrics about the state of your cluster out of the box! These metrics can be used to serve as an insight into your cluster: Either through metrics alone, in the form of dashboards, or through an alerting pipeline. They can especially be useful to troubleshoot an outage. A metric worth pointing out is `kube_pod_container_status_restarts_total` which can be used to alert you on a crashing pod. Another helpful metric is `kube_deployment_status_replicas` which together with `kube_deployment_status_replicas_available` can be used to alert on whether a deployment is rolled out successfully or stuck. And there are many more metrics available! If you are interested in further metrics and their details, check out the [documentation](https://github.com/kubernetes/kube-state-metrics/tree/master/docs#documentation).
+
+## What is new in v2.0?
+
+So now that we know what kube-state-metrics is, we are excited to announce the next release: kube-state-metrics v2.0! This release was long-awaited and started with an alpha release in September 2020. To ease maintenance we removed tech debt and also adjusted some confusing wording around user-facing flags and APIs. We also removed some metrics that caused unnecessarily high cardinality in Prometheus! For the 2.0 release, we took the time to set up scale and performance testing. This allows us to better understand if we hit any issues in large clusters and also to document resource request recommendations for your clusters. In this release (and v1.9.8) container builds providing support for multiple architectures were introduced allowing you to run kube-state-metrics on ARM, ARM64, PPC64 and S390x as well!
+
+The following is a list of the most impactful user-facing breaking changes. The full list of 2.0 release changes includes other breaking changes regarding metrics dropped or renamed, along with new metrics and features added, for that see the full [CHANGELOG](https://github.com/kubernetes/kube-state-metrics/blob/master/CHANGELOG.md) file and the end.
+
+So without further ado, here is the list of more noteworthy user-facing breaking changes. A full list of changes, features and bug fixes is available in the changelog at the end of this post.
```

Comment:
Have any port number defaults changed in this release?

Reply:
None in this release.